src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -695,6 +706,98 @@ internal static string ReadStringOrNumberAsString(ref Utf8JsonReader reader, str
             return retVal;
         }
 
+        internal const string TryToCreateDateTimeClaimsSwitch = "Switch.Microsoft.IdentityModel.TryAllStringClaimsAsDateTime";
+
+        public static bool TryAllStringClaimsAsDateTime()
+        {
+            return (AppContext.TryGetSwitch(TryToCreateDateTimeClaimsSwitch, out bool tryAsDateTime) && tryAsDateTime);
+        }
+
+        /// <summary>
+        /// This is a non-exhaustive list of claim types that are not expected to be DateTime values
+        /// sourced from expected Entra V1 and V2 claims, OpenID Connect claims, and a selection of
+        /// restricted claim names.
+        /// </summary>
+        public static HashSet<string> KnownNonDateTimesClaimTypes = new(StringComparer.Ordinal)
```

Comment:
There may be a better type to put this on, open to suggestions

Reply:
I think this can be internal.
The switch will handle backcompat.

Reply:
I'll put this behind a method, no need for the Collection to be exposed

Reply:
These are all from external sources, I think it is more important to be able to read the values exactly in this case vs having them in constants elsewhere. I'd agree if they got uses elsewhere.